packages/core/utils/src/payment/abstract-payment-provider.ts

```diff
@@ -623,6 +625,56 @@ export abstract class AbstractPaymentProvider<TConfig = Record<string, unknown>>
     context: UpdatePaymentProviderSession
   ): Promise<PaymentProviderError | PaymentProviderSessionResponse>
 
+  /**
+   * List the payment methods associated with the context (eg. customer) of the payment provider, if any.
+   *
+   * @param context - The context for which the payment methods are listed. Usually the customer should be provided.
+   * @returns An object whose `payment_methods` property is set to the data returned by the payment provider.
+   *
+   * @example
+   * // other imports...
+   * import {
+   *   PaymentProviderContext,
+   *   PaymentProviderError,
+   *   PaymentMethodResponse
+   *   PaymentProviderSessionResponse,
+   * } from "@medusajs/framework/types"
+   *
+   *
+   * class MyPaymentProviderService extends AbstractPaymentProvider<
+   *   Options
+   * > {
+   *   async listPaymentMethods(
+   *     context: PaymentProviderContext
+   *   ): Promise<PaymentMethodResponse> {
+   *     const {
+   *       customer,
+   *     } = context
+   *     const externalCustomerId = customer.metadata.stripe_id
+   *
+   *     try {
+   *       // assuming you have a client that updates the payment
+   *       const response = await this.client.listPaymentMethods(
+   *         {customer: externalCustomerId}
+   *       )
+   *
+   *       return {payment_methods: response}
+   *     } catch (e) {
+   *       return {
+   *         error: e,
+   *         code: "unknown",
+   *         detail: e
+   *       }
+   *     }
+   *   }
```

Comment:
suggestion: I think this need some update based on the latest API impl

Reply:
Yes nice catch!